src/libsyntax/parse/parser.rs

```diff
@@ -2072,8 +2072,15 @@ impl<'a> Parser<'a> {
                     start: Option<P<Expr>>,
                     end: Option<P<Expr>>,
                     limits: RangeLimits)
-                    -> ast::ExprKind {
-        ExprKind::Range(start, end, limits)
+                    -> PResult<'a, ast::ExprKind> {
+        if end.is_none() && limits == RangeLimits::Closed {
+            Err(self.span_fatal_help(self.span,
+                                     "inclusive range with no end",
+                                     "currently, inclusive ranges must be bounded at the end \
+                                      (`...b` or `a...b`) -- see tracking issue #28237"))
```

Comment:
I don't think we often include issue numbers in error messages. If there is something specific we want feedback on, then it might make sense, but in that case it needs a url rather than a number and much more explanation.